azurelinuxagent/ga/update.py

```diff
@@ -487,9 +488,44 @@ def _process_goal_state(self, exthandlers_handler, remote_access_handler):
         finally:
             self.last_incarnation = incarnation
 
+    @staticmethod
+    def __get_vmagent_upgrade_status(protocol, incarnation_changed):
+        """
+        This function gets the VMAgent upgrade status as per the last GoalState.
+        Returns: None if the last GS does not ask for requested version else VMAgentUpdateStatus
+        """
+        upgrade_status = None
+
+        try:
+            agent_manifests, _ = protocol.get_vmagent_manifests()
+            # Expectation here is that there will only be one manifest per family passed down from CRP.
+            # If there are multiple, we pick the first one
+            manifest = next(m for m in agent_manifests if m.family == conf.get_autoupdate_gafamily())
+            if manifest.is_requested_version_available:
+                if CURRENT_VERSION == manifest.requested_version:
+                    status = VMAgentUpdateStatuses.Success
+                    code = 0
```

Comment:
Its used for logging in CRP. Windows agent primarily sends different HR Codes for different cases which gets logged in CRP. Here's what Aymen said about it - 

> Usually it would be an error code(mainly for windows)
Its only for logging
CRP will only cares about one thing : "status"

Reply:
if crp does not care about it, why populate it at all? or we need some documentation about what 0 and -1 mean

Reply:
CRP logs it. I think the intention is to have a clear message on what went wrong in the CRP logs itself. 

>  why populate it at all?

It's primarily needed for Windows where they're going to return different HR codes based on the scenario.   

> or we need some documentation about what 0 and -1 mean

Since we dont have a specific need for it, I just stuck to the basic exit code usages that we generally use in the code (0 for success and -1 for failure). Let me know if you feel the need to have a comment about it and I can add it.

Reply:
in what instances do we use -1 for failure?

Reply:
You're right, the normal exit codes are positive non-0s. I'll change it to `1`

Reply:
thanks